src/openrct2/GameState.h

```diff
@@ -30,6 +30,7 @@ namespace OpenRCT2
         ClimateState ClimateNext;
         money64 Cash;
         money64 InitialCash;
+        bool WindowSceneryScatterEnabled = false;
```

Comment:
I could not find any other way to reference this variable other than using the gameState. I will change this if I can find another way.

Reply:
Anything that is UI is not part of the game state. The game action needs to be refactored to not propagate the error when just a single placement fails.

Reply:
Do you think that I should create a struct similar to GameState but holds all of the booleans that track which windows are open. I think that it would greatly help to fix issues like these which are specific to one windows. I could move all booleans into this struct in this PR.

Reply:
I think its best not to brute force the solution, have a look at the code that is invoking the game action, I think its located in TopToolbar.cpp or something like that, it currently invokes game actions in a loop to achieve the scatter functionality which is not ideal like that. I think the best way to make this work better is to have a new game action for scatter rather than using game actions within a loop like it currently does, its also not very practical for network play as it will send potentially hundreds of game actions to the server.